src/App.vue

```diff
@@ -357,20 +389,27 @@ export default {
     /** Fetches and stores the business data. */
     async fetchBusinessData (): Promise<void> {
       const data = await Promise.all([
+        // FUTURE: all of these should be store actions
         AuthServices.fetchEntityInfo(this.getAuthApiUrl, this.businessId),
-        LegalServices.fetchBusinessInfo(this.businessId),
+        this.loadBusinessInfo(this.businessId),
         LegalServices.fetchTasks(this.businessId),
         LegalServices.fetchFilings(this.businessId || this.tempRegNumber),
         LegalServices.fetchParties(this.businessId)
       ])
 
       if (!data || data.length !== 5) throw new Error('Incomplete business data')
 
+      // store data from calls above
       this.storeEntityInfo(data[0])
-      await this.storeBusinessInfo(data[1])
       this.storeTasks(data[2])
       this.storeFilings(data[3])
       this.storeParties(data[4])
+
+      // if the URL was provided in business info, load state filing
+      if (this.getStateFilingUrl) await this.loadStateFiling(this.getStateFilingUrl)
+
+      // now that we know entity type, store config object
+      this.storeConfigObject()
```

Comment:
Both of these were previously done inside storeBusinessInfo() (which is now a store action). They depend on business info data, and I didn't want them in the store action (for now), so I moved them here.